test/built-ins/Map/prototype/constructor/is-map.js

```diff
@@ -0,0 +1,10 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 23.1.3.2
+description: >
+  The initial value of Map.prototype.constructor is the intrinsic object %Map%.
+---*/
+
+assert.sameValue(Map.prototype.constructor, Map);
+assert.sameValue((new Map()).constructor, Map);
```

Comment:
Could you check the property descriptor here, too? Once that's done, you can probably nix the directory and just define `constructor.js` (in the same way you're doing with `Symbol.iterator.js`. I guess that file organization is a little inconsistent with the existing tests for Map, so I think it would be okay to do in a follow-up patch.